private/buf/bufformat/formatter.go

```diff
@@ -405,10 +430,10 @@ func (f *formatter) writeOptionPrefix(optionNode *ast.OptionNode) {
 //	(custom.thing).bridge.(another.thing)
 func (f *formatter) writeOptionName(optionNameNode *ast.OptionNameNode) {
 	for i := 0; i < len(optionNameNode.Parts); i++ {
-		if i == 0 {
-			// The comments will have already been written as a multiline
-			// comment above the option name, so we need to handle this case
-			// specially.
+		if f.inCompactOptions && i == 0 {
```

Comment:
Previously, the option name rarely had a leading comment, which is why this bug was never noticed before.

We've only written all of the leading comments for compact options. Now that the new protocompile AST is likely to contain leading comments here, this bug was noticed (by seeing the formatter completely _drop_ a comment for a non-compact option declaration... oops!).